### PR TITLE
[CP Stag] Add GBR to pending wallet reports & update perview for 1:1 reports

### PR DESCRIPTION
**Pull request by b4s36t4**
### Details

This PR fixes the issue with not showing GBR for the report that require user action to complete create wallet and as well to show correct preview message for the 1:1 chat instead of showing system message.

1. https://github.com/Expensify/App/issues/30033
2. https://github.com/Expensify/App/issues/30037

### Fixed Issues

$ https://github.com/Expensify/App/issues/30037
PROPOSAL: https://github.com/Expensify/App/issues/30037#issuecomment-1772395813 

$ https://github.com/Expensify/App/issues/30033
PROPOSAL: https://github.com/Expensify/App/issues/30033#issuecomment-1772477889

### Tests

- [x] Verify that no errors appear in the JS console

1. Open App request any amount form any user (Silver Wallet)
2. Observe the 1:1 chat and IOU report both should show owes message.
3. Pay money using expensify method from User B (gold wallet)
4. For User B there shouldn't be no GBR
5. For User A there should GBR on IOU report
6. For 1:1 the preview should `paid` message
7. For IOU report the message should contain paid & amount held system message both should be there

### Offline tests

Same as above
### QA Steps

- [x] Verify that no errors appear in the JS console
Same as above
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/59088937/ecb396d9-9e27-4b6a-ad4a-adf61bf75b9f

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/59088937/46bdcbeb-a8fb-43ed-9d47-25e55f4f5c70

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/59088937/0fd8f630-6bdd-4415-acd4-0d52688643ea

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/59088937/fc72f143-7b35-4182-b7d0-b0fbbe03ccfe

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/59088937/9f60e8ca-2c8c-4f5b-89f6-4b1da06f23ca

Screenshot 2023-10-20 at 8 41 39 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/59088937/d9683d1c-3893-41c4-97b6-e99add24755c

</details>

### CODE REVIEW

**Comment by b4s36t4** [comment]
@situchan @techievivek PR is up.!

**Comment by situchan** [comment]
@b4s36t4 I think that's a good test case. We can update it to be compatible for our change.

**Comment by situchan** [comment]
Also, please watch https://github.com/Expensify/App/pull/30079 and pull main after it's merged.

**Comment by luacmartins** [comment]
Agree with @situchan let's update the test

**Comment by techievivek** [comment]
@b4s36t4 Yes, let's please update the tests as per new changes rather than getting rid of it.

**Comment by techievivek** [comment]
@situchan Can you please give this a review, I will also test this on my end.

**Comment by situchan** [comment]
@techievivek unit test failing

File: src/libs/ReportUtils.js
Comment (techievivek) [inline src/libs/ReportUtils.js:1299]:
```suggestion
    // hasOutstandingIOU will be false if the user paid, but isWaitingOnBankAccount will be true if user don't have a wallet or bank account setup
```

File: src/libs/OptionsListUtils.js
Comment (techievivek) [inline src/libs/OptionsListUtils.js:401]:
```suggestion
        // Whether parent report is a 1:1 report
```

File: src/libs/ReportUtils.js
Comment (b4s36t4) [reply src/libs/ReportUtils.js:278]: I tried `isChatReport` it's giving eslint error as there's an already existing function is same file.

Comment (situchan) [reply src/libs/ReportUtils.js:278]: Introducing 2nd param changes the meaning of `isSettled`.
So it's a bit odd that settled if chat report, not settled if IOU report.
Can we think of better way without altering the nature of this function?

Comment (b4s36t4) [reply src/libs/ReportUtils.js:278]: @situchan then I think we need to update the condition to moved inside `getReportPreviewMessage` function. Would that work?!

File: src/libs/OptionsListUtils.js
Comment (techievivek) [inline src/libs/OptionsListUtils.js:402]: Should we just add a condition here rather than passing the param all the way to `isSettled`?

Like if the iouReport has `isWaitingOnBankAccount` set and the current report is a DM? And let's create a separate function for showing settled message?

```
        // A settled report preview message can come in three formats "paid ... elsewhere" or "paid ... with Expensify"
        let translatePhraseKey = 'iou.paidElsewhereWithAmount';
        if (
            _.contains([CONST.IOU.PAYMENT_TYPE.VBBA, CONST.IOU.PAYMENT_TYPE.EXPENSIFY], lodashGet(reportAction, 'originalMessage.paymentType')) ||
            reportActionMessage.match(/ (with Expensify|using Expensify)$/)
        ) {
            translatePhraseKey = 'iou.paidWithExpensifyWithAmount';
        }
        return Localize.translateLocal(translatePhraseKey, {amount: formattedAmount, payer: payerName});
```

Comment (techievivek) [reply src/libs/OptionsListUtils.js:402]: @situchan ^ because I saw your comment regarding new code in `isSettled` method.

Comment (b4s36t4) [reply src/libs/OptionsListUtils.js:402]: @techievivek won't this be duplicate of same logic again? Please check I have updated the condition to not to include variables inside `isSettled` function.

File: src/libs/ReportUtils.js
Comment (b4s36t4) [inline src/libs/ReportUtils.js:1683]: @situchan @techievivek review this please. Updated to resolve issue with `isSettled` function update.

**Comment by b4s36t4** [comment]
https://github.com/Expensify/App/assets/59088937/b5308d52-428b-4bf0-8135-996c59c6a5eb

> With both checks for expense & chat reports (1:1 & IOU thread)

File: src/libs/ReportUtils.js
Comment (techievivek) [reply src/libs/ReportUtils.js:1683]: Would it not show as settled for the iou report now?

File: src/libs/ReportUtils.js
Comment (techievivek) [inline src/libs/ReportUtils.js:286]: Personal policies are only used for iou reports. For expense report we always use group policies(free, control or collect policy) and we add the status as APPROVED.

So we can update the comment to say 

// APPROVED status is for free group policy

File: src/libs/ReportUtils.js
Comment (b4s36t4) [reply src/libs/ReportUtils.js:1683]: That's a or condition so it would show paid for iou as well.

File: src/libs/ReportUtils.js
Comment (b4s36t4) [reply src/libs/ReportUtils.js:286]: Would update this.

File: src/libs/ReportUtils.js
Comment (techievivek) [reply src/libs/ReportUtils.js:1683]: That's what I am saying then how would it show system message in case of iou report.

File: src/libs/ReportUtils.js
Comment (situchan) [inline src/libs/ReportUtils.js:282]: Why ` || report.isWaitingOnBankAccount` is removed here? This makes all pending requests as settled

Before:
Screenshot 2023-10-23 at 2 35 55 PM



After:
Screenshot 2023-10-23 at 2 36 29 PM

Comment (techievivek) [reply src/libs/ReportUtils.js:282]: Yeah, we just resolved this in DM, I pointed that out here: https://github.com/Expensify/App/pull/30090/files#r1368314158

**Comment by situchan** [comment]
How about introducing new function called `isPaid`?
So isSettled = isPaid && isWaitingOnBankAccount === false

**Comment by techievivek** [comment]
> How about introducing new function called isPaid?
So isSettled = isPaid && isWaitingOnBankAccount === false

The issue here is that for DM chat report that contains the IOU preview we want to show the settled message in LHN even if there is `isWaitingOnBankAccount` property set on the IOU report, but for the IOU report, we want to show the system message that we are waiting for the payee to enable their wallet/bank account.

**Comment by techievivek** [comment]
We can keep the `isSettled` logic unchanged and then we can add an OR operator to handle the above case.

```
if (isSettled(report.reportID) || (report.isWaitingOnBankAccount && isPreviewMessageForParentChatReport)) {
    // Show the paid message
}
```

**Comment by situchan** [comment]
`isSettled` is used in many places. i.e. MoneyRequestHeader
I believe we should not modify any request if paid and isWaitingOnBankAccount.
But as seen here, it's still editable:
https://github.com/Expensify/App/blob/e73cd31a567c5f026f3a9f1435ecb6f99b2fae82/src/pages/EditRequestPage.js#L118-L122

**Comment by techievivek** [comment]
> I believe we should not modify any request if paid and isWaitingOnBankAccount.

Yeah, I agree. Even if we're not adding this right now, we should update it as soon as possible.

**Comment by b4s36t4** [comment]
@situchan @techievivek Updated the condition here https://github.com/Expensify/App/pull/30090/files#diff-65c096044d5f69b35bcdec14c99c4fda4580759df9b1a7c36650d58eea276f1dR1682-R1683

Sorry for back&forth changes :)

File: src/libs/ReportUtils.js
Comment (b4s36t4) [reply src/libs/ReportUtils.js]: Done, fixed.

File: src/libs/OptionsListUtils.js
Comment (situchan) [inline src/libs/OptionsListUtils.js:401]: This comment is not necessary. Also, this involves workspace chat as well, not only 1:1

File: src/libs/OptionsListUtils.js
Comment (situchan) [inline src/libs/OptionsListUtils.js:402]: No need to introduce new variable. inline is better

File: src/libs/ReportUtils.js
Comment (situchan) [inline src/libs/ReportUtils.js:1640]: Same here. Also involves workspace chat

File: src/libs/OptionsListUtils.js
Comment (b4s36t4) [reply src/libs/OptionsListUtils.js]: Fixed.

File: src/libs/OptionsListUtils.js
Comment (b4s36t4) [reply src/libs/OptionsListUtils.js]: Updated to inline.

**Comment by situchan** [comment]
This is minor inconsistency:
```
Started settling up, payment is held until ${submitterDisplayName} enables their Wallet
started settling up, payment is held until ${submitterDisplayName} adds a bank account
```

Can we lowercase first letter for wallet message?

**Comment by b4s36t4** [comment]
@situchan lowercase for wallet message or upper case for bank message? @techievivek for review :)

**Comment by situchan** [comment]
> @situchan lowercase for wallet message or upper case for bank message? @techievivek for review :)

I believe lowercase. All system messages already do.

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details open>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/18078f5d-4e4b-4a9b-a75f-eff5fcf9e0c2


https://github.com/Expensify/App/assets/108292595/dec16eb5-6b0f-490b-aa89-b456a5ca2a01



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by situchan** [comment]
I just noticed that LHN last message is wrong. It says "paid elsewhere"

Screenshot 2023-10-23 at 4 25 17 PM

**Comment by b4s36t4** [comment]
@situchan  Checking.

**Comment by techievivek** [comment]
The RCA of the above bug is that we don't add the PAY IOU action on the expense report if the payee don't have a linked bank account: https://github.com/Expensify/Auth/blob/40dc46bab88a5c1a45edf0b2e4cdd02808ca59e1/auth/command/PayMoneyRequest.cpp#L291, as a result we are unable to fetch the correct report action that would tell us the correct payment type.

**Comment by techievivek** [comment]
CC @MariaHCD @mountiny please have a look at the above comment https://github.com/Expensify/App/pull/30090#issuecomment-1775032232 the flow here is significantly different from the PayWithWallet flow. (Honestly speaking there is a lot of cleanup things that are there on the flow and maybe we should revisit the whole money request/pay flow and make it consistent).

**Comment by b4s36t4** [comment]
@situchan Please check now, I have fixed the `paid elsewhere` issue.

**Comment by techievivek** [comment]
@situchan Can we have another review here, please? 🙏

File: src/libs/ReportUtils.js
Comment (situchan) [inline src/libs/ReportUtils.js:1694]: Can you make sure that this OR condition doesn't cause regression of showing `paid with Expensify` for even non paid action?

Comment (b4s36t4) [reply src/libs/ReportUtils.js:1694]: Non-paid action in-the sense?

Comment (situchan) [reply src/libs/ReportUtils.js:1694]: Updated comment.

Comment (b4s36t4) [reply src/libs/ReportUtils.js:1694]: Yes, I think `isSettled` would do the work.

Comment (situchan) [reply src/libs/ReportUtils.js:1694]: ok, so ideally this should be fixed in backend. Given urgency, frontend hot-fix would be fine here.

**Comment by situchan** [comment]
So far looks good against ngrok server.

I will re-point to production server and retest IOU, expense request with elsewhere pay (without wallet)

**Comment by situchan** [comment]
I'm seeing GBR even though I am not payer but payee. I have nothing to pay

Screenshot 2023-10-23 at 6 33 17 PM

File: src/libs/ReportUtils.js
Comment (situchan) [inline src/libs/ReportUtils.js:1325]: This condition is still wrong. It causes regression above ^

Comment (b4s36t4) [reply src/libs/ReportUtils.js:1325]: I don't think so, I removed it and tested it still shows.

Comment (situchan) [reply src/libs/ReportUtils.js:1325]: I am not seeing GBR after reverting this condition.

Comment (b4s36t4) [reply src/libs/ReportUtils.js:1325]: @situchan I think that dot is intentional, it's not showing in main (tested) because the condition is wrong (the RCA for GBR bug). Maybe you're trying in a new account.

> // Money request waiting for current user to add their credit bank account

This feels like intentional.

Comment (b4s36t4) [reply src/libs/ReportUtils.js:1325]: > I am not seeing GBR after reverting this condition

Means `report.hasOutstandingIOU && report.ownerAccountID === currentUserAccountID` this would be the conditon right?!

Comment (situchan) [reply src/libs/ReportUtils.js:1325]: you haven't fully reverted code.
`report.isWaitingOnBankAccount` is missing

**Comment by mountiny** [comment]
@techievivek I dont necessarily think that is a problem (though I agree that the fact these two flows are inconsistent is not good, the wallet flow was added as second as the reimbursement flow already existed in the oldDot)

I think we should mainly not be showing "paid this and this" when the money actually has not left their account yet, nothing will happen until the employee adds the bank account and in that case its fine there is no Pay action and the alternate text should more so talk about hte fact the user needs to add a bank account imho

File: src/libs/ReportUtils.js
Comment (b4s36t4) [reply src/libs/ReportUtils.js]: @situchan Fixed, Can you please check?

**Comment by Julesssss** [comment]
Failing test looks related to changes: 

Screenshot 2023-10-23 at 14 26 05

**Comment by b4s36t4** [comment]
@Julesssss fixed.

**Comment by techievivek** [comment]
@mountiny 
> I dont necessarily think that is a problem (though I agree that the fact these two flows are inconsistent is not good, the wallet flow was added as second as the reimbursement flow already existed in the oldDot)

My primary concern is how some of our code, particularly on the frontend, tries to follow the same code for both flows. At the same time, in the backend, we have different commands handling this differently, resulting in inconsistencies. For e.g. in the case when the user needs to add a bank account or wallet. So we should either try to make it consistent or handle it distinctly.

I think we can re-look into the code(both frontend and backend) and try to clean this up once the urgency has settled a bit.

**Comment by mountiny** [comment]
> I think we can re-look into the code(both frontend and backend) and try to clean this up once the urgency has settled a bit.

I agree

File: src/libs/ReportUtils.js
Comment (mountiny) [inline src/libs/ReportUtils.js:286]: Type
```suggestion
    // In case the payment is scheduled and we are waiting for the payee to set up their wallet,
```

File: src/libs/ReportUtils.js
Comment (mountiny) [inline src/libs/ReportUtils.js:1256]:
```suggestion
```

**Review by Julesssss — APPROVED** [review]
LGTM with the latest changes, I'll leave final review to @mountiny

**Comment by situchan** [comment]
Elsewhere flow looks good.
Currently retesting with gold wallet flow as GBR condition changed

**Review by mountiny — APPROVED** [review]
Thanks!

**Comment by mountiny** [comment]
@situchan let me know in slack once you finished, thanks!

**Comment by situchan** [comment]
@b4s36t4 did you retest gold wallet flow on payer side? works fine on payee side.
@techievivek stopped ngrok so not able to test.

**Comment by b4s36t4** [comment]
Yes, on payer side it will show GBR till user pays, it will hide after payment done.

> payer side is verified when connected to ngork, payee side verified without ngrok (ngrok is slow & taking long to get response)

**Review by situchan — APPROVED** [review]
Approving as @b4s36t4 covered one test case I missed.
All other cases work well.

**Pull request merged**
